File: DanmakuFlameMaster/src/main/java/master/flame/danmaku/danmaku/model/DanmakuTimer.java
/*
 * Copyright (C) 2013 Chen Hui <calmer91@gmail.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package master.flame.danmaku.danmaku.model;

public class DanmakuTimer {
    protected long currMillisecond;

    protected long lastInterval;

    public DanmakuTimer() {

    }

    public DanmakuTimer(long curr) {
        update(curr);
    }

    public long update(long curr) {
        lastInterval = curr - currMillisecond;
        currMillisecond = curr;
        return lastInterval;
    }

    public long add(long mills) {
        return update(currMillisecond + mills);
    }

    public long lastInterval() {
        return lastInterval;
    }

    public long currMillisecond() { return currMillisecond; }

}

File: danmaku-engine/src/commonMain/kotlin/cn/a10miaomiao/bilimiao/danmaku/model/DanmakuTimer.kt
package cn.a10miaomiao.bilimiao.danmaku.model

open class DanmakuTimer {

    var currMillisecond: Long = 0L
        protected set

    var lastInterval: Long = 0L
        protected set

    constructor()

    constructor(curr: Long) {
        update(curr)
    }

    open fun update(curr: Long): Long {
        lastInterval = curr - currMillisecond
        currMillisecond = curr
        return lastInterval
    }

    fun add(mills: Long): Long {
        return update(currMillisecond + mills)
    }
}
